versions/draft-3/SPEC.md

```diff
@@ -3053,6 +3055,30 @@ Array[String] zs = [ "d", "e" ]
 Array[Pair[Int, String]] crossed = cross(xs, zs) # i.e. crossed = [ (1, "d"), (1, "e"), (2, "d"), (2, "e"), (3, "d"), (3, "e") ]
 ```
 
+## Array[Pair[X,Y]] as_pairs(Map[X,Y])
+
+Given any two Object types in a Map, the `as_pairs` function returns an Array containing each element in the form of a Pair. The key will be the left element of the Pair and the value the right element.
+
+```
+Map[String, Int] x = {"a": 1, "b": 2, "c": 3}
+Map[String,Pair[File,File] y = {"a": ("a.bam, "a.bai"), "b": ("b.bam", "b.bai")}
+
+Array[Pair[String,Int]] xpairs = as_pairs(x) # [("a", 1), ("b", 2), ("c", 3)]
+Array[Pair[String,Pair[File,File]] ypairs = as_pairs(y) # [("a", ("a.bam", "a.bai")), ("b", ("b.bam", "b.bai"))]
+```
+
+## Map[X,Y] as_map(Array[Pair[X,Y]])
+
+Given any two Object types in an Array of Pairs, the `as_map` function returns a Map in which the given left element is the key and the right element the value.
```

Comment:
Could you add a comment on what happens if there's a key collision, eg `[(1, 2), (1, 3)]`?